src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,371 @@
+                ])
+                    .then(([googlePay, googlePayPaymentDataRequest]) => {
+                        this._googlePaymentsClient = this._getGooglePaymentsClient(googlePay, testMode);
+                        this._googlePaymentDataRequest = googlePayPaymentDataRequest;
+                    })
+                    .catch((error: Error) => {
+                        this._handleError(error);
+                    });
+            });
+    }
+
+    private _displayWallet(): Promise<InternalCheckoutSelectors> {
+        return new Promise((resolve, reject) => {
+            if (!this._paymentMethod) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            if (!this._googlePaymentsClient && !this._googlePaymentDataRequest) {
+                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            }
+
+            this._googlePaymentsClient.isReadyToPay({
+                allowedPaymentMethods: this._googlePaymentDataRequest.allowedPaymentMethods,
+            }).then( (response: GooglePayIsReadyToPayResponse) => {
+                if (response) {
+                    this._googlePaymentsClient.loadPaymentData(this._googlePaymentDataRequest)
+                        .then((paymentData: GooglePaymentData) => {
+                            return this._setExternalCheckoutData(paymentData);
+                        }).catch((err: GooglePaymentsError) => {
+                        reject(new Error(err.statusCode));
+                    });
+                }
+            });
+        });
+    }
+
+    private _createOrder(payment: Payment, useStoreCredit?: boolean, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._orderActionCreator.submitOrder({ useStoreCredit }, options))
+            .then(() => this._store.dispatch(this._paymentActionCreator.submitPayment(payment)));
+    }
+
+    private _getGooglePaymentsClient(google: GooglePaySDK, testMode: boolean | undefined): GooglePayClient {
+        let environment: EnvironmentType;
+        testMode = true; // TODO: remove when push this code to final review
+        if (testMode === undefined) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        if (!testMode) {
+            environment = 'PRODUCTION';
+        } else {
+            environment = 'TEST';
+        }
+
+        const options: GooglePayPaymentOptions = { environment };
+
+        return new google.payments.api.PaymentsClient(options);
+    }
+
+    private _setExternalCheckoutData(paymentData: GooglePaymentData): Promise<void> {
+        return this._googlePayInitializer.parseResponse(paymentData)
+            .then((tokenizePayload: TokenizePayload) => {
+                const paymentSuccessPayload: PaymentSuccessPayload = {
```

Comment:
Is there a reason why you need to declare `PaymentSuccessPayload`? The type can be inferred and if you have a missing property, the compiler will complain when you call `_updateShippingAndBillingAddress` / `_paymentInstrumentSelected` so it's not really necessary to declare it again.